Index: GaudiKernel/IScheduler.h
===================================================================
--- GaudiKernel/IScheduler.h	(revision 7911)
+++ GaudiKernel/IScheduler.h	(working copy)
@@ -8,6 +8,7 @@
 // C++ include files
 #include <functional>
 #include <vector>
+#include <string>
 
 /**@class IScheduler IScheduler.h GaudiKernel/IScheduler.h
  *
@@ -32,6 +33,10 @@
   /// Retrieve a finished event from the scheduler
   virtual StatusCode popFinishedEvent(EventContext*& eventContext) = 0;
 
+  //HACK
+  //virtual std::string BENBENBEN() = 0;
+  virtual StatusCode BENBENBEN( std::string AlgName, EventContext* eventContext ) = 0;
+
   /// Try to retrieve a finished event from the scheduler
   virtual StatusCode tryPopFinishedEvent(EventContext*& eventContext) = 0;
   
Index: GaudiKernel/EventContext.h
===================================================================
--- GaudiKernel/EventContext.h	(revision 7911)
+++ GaudiKernel/EventContext.h	(working copy)
@@ -38,6 +38,10 @@
     m_valid = (e<0 || s == INVALID_CONTEXT_ID) ? false: true;
   }
 
+  virtual ~EventContext()
+  {
+  }
+
   ContextEvt_t evt() const { return m_evt_num; }
   ContextID_t slot() const { return m_evt_slot; }
   bool valid() const {return m_valid;}
