diff -ur bzip2-1.0.5-orig/Makefile bzip2-1.0.5/Makefile
--- bzip2-1.0.5-orig/Makefile	2008-02-14 13:39:18.000000000 +0100
+++ bzip2-1.0.5/Makefile	2009-01-30 15:12:37.000000000 +0100
@@ -18,13 +18,13 @@
 SHELL=/bin/sh
 
 # To assist in cross-compiling
-CC=gcc
+CC=$(MY_CC)
 AR=ar
 RANLIB=ranlib
-LDFLAGS=
+LDFLAGS=$(MY_LDFLAGS)
 
 BIGFILES=-D_FILE_OFFSET_BITS=64
-CFLAGS=-Wall -Winline -O2 -g $(BIGFILES)
+CFLAGS=-Wall -Winline -O2 $(BIGFILES) $(MY_CFLAGS)
 
 # Where you want it installed when you do 'make install'
 PREFIX=/usr/local
diff -ur bzip2-1.0.5-orig/Makefile-libbz2_so bzip2-1.0.5/Makefile-libbz2_so
--- bzip2-1.0.5-orig/Makefile-libbz2_so	2007-12-09 14:00:50.000000000 +0100
+++ bzip2-1.0.5/Makefile-libbz2_so	2009-01-30 15:14:16.000000000 +0100
@@ -22,9 +22,10 @@
 
 
 SHELL=/bin/sh
-CC=gcc
+CC=$(MY_CC)
 BIGFILES=-D_FILE_OFFSET_BITS=64
-CFLAGS=-fpic -fPIC -Wall -Winline -O2 -g $(BIGFILES)
+CFLAGS=-fpic -fPIC -Wall -Winline -O2 $(BIGFILES) $(MY_CFLAGS)
+LDFLAGS=$(MY_LDFLAGS)
 
 OBJS= blocksort.o  \
       huffman.o    \
@@ -35,13 +36,13 @@
       bzlib.o
 
 all: $(OBJS)
-	$(CC) -shared -Wl,-soname -Wl,libbz2.so.1.0 -o libbz2.so.1.0.4 $(OBJS)
-	$(CC) $(CFLAGS) -o bzip2-shared bzip2.c libbz2.so.1.0.4
+	$(CC) $(LDFLAGS) -shared -o libbz2.so.1.0.5 $(OBJS)
+	$(CC) $(CFLAGS) -o bzip2-shared bzip2.c libbz2.so.1.0.5
 	rm -f libbz2.so.1.0
-	ln -s libbz2.so.1.0.4 libbz2.so.1.0
+	ln -s libbz2.so.1.0.5 libbz2.so.1.0
 
 clean: 
-	rm -f $(OBJS) bzip2.o libbz2.so.1.0.4 libbz2.so.1.0 bzip2-shared
+	rm -f $(OBJS) bzip2.o libbz2.so.1.0.5 libbz2.so.1.0 bzip2-shared
 
 blocksort.o: blocksort.c
 	$(CC) $(CFLAGS) -c blocksort.c
