--- pyAMI/object.py-orig	2020-01-07 21:33:33.000000001 +0100
+++ pyAMI/object.py	2020-01-07 21:33:51.000000001 +0100
@@ -144,7 +144,7 @@
 
 		for rowset in self.rowsets:
 
-			if not rowset_type or (rowset.attributes.has_key('type') and rowset.attributes['type'].value == rowset_type):
+			if not rowset_type or ('type' in rowset.attributes.keys() and rowset.attributes['type'].value == rowset_type):
 
 				for row in rowset.getElementsByTagName('row'):
 					field_dict = pyAMIDict()
@@ -152,7 +152,7 @@
 					for field in row.getElementsByTagName('field'):
 						name = field.attributes['name'].value
 
-						if field.attributes.has_key('table'):
+						if 'table' in field.attributes.keys():
 							table = field.attributes['table'].value
 
 							if self.entity and table and self.entity != table:
