diff -urN lapack-3.1.1-orig/Makefile lapack-3.1.1/Makefile
--- lapack-3.1.1-orig/Makefile	2009-06-18 20:04:32.000000000 +0200
+++ lapack-3.1.1/Makefile	2009-06-18 20:23:03.000000000 +0200
@@ -14,8 +14,7 @@
 clean: cleanlib cleantesting cleanblas_testing 
 
 lapack_install:
-	( cd INSTALL; $(MAKE); ./testlsame; ./testslamch; \
-	  ./testdlamch; ./testsecond; ./testdsecnd; ./testversion )
+	( cd INSTALL; $(MAKE)  )
 
 blaslib:
 	( cd BLAS/SRC; $(MAKE) )
diff -urN lapack-3.1.1-orig/SRC/Makefile lapack-3.1.1/SRC/Makefile
--- lapack-3.1.1-orig/SRC/Makefile	2009-06-18 20:04:32.000000000 +0200
+++ lapack-3.1.1/SRC/Makefile	2009-06-18 20:04:57.000000000 +0200
@@ -294,7 +294,7 @@
 	$(ALLAUX)
 
 ../$(LAPACKLIB): $(ALLOBJ)
-	$(ARCH) $(ARCHFLAGS) $@ $(ALLOBJ)
+	$(ARCH) $(ARCHFLAGS) $@ $(ALLOBJ) $(BLASLIB)
 	$(RANLIB) $@
 
 single: $(SLASRC) $(ALLAUX) $(SCLAUX) 
diff -urN lapack-3.1.1-orig/TESTING/MATGEN/Makefile lapack-3.1.1/TESTING/MATGEN/Makefile
--- lapack-3.1.1-orig/TESTING/MATGEN/Makefile	2009-06-18 20:04:32.000000000 +0200
+++ lapack-3.1.1/TESTING/MATGEN/Makefile	2009-06-18 20:05:37.000000000 +0200
@@ -60,7 +60,7 @@
 
 ../../$(TMGLIB): $(SMATGEN) $(CMATGEN) $(SCATGEN) $(DMATGEN)	\
 		$(ZMATGEN) $(DZATGEN)
-	$(ARCH) $(ARCHFLAGS) $@ $(ALLOBJ)
+	$(ARCH) $(ARCHFLAGS) $@ $(ALLOBJ) $(BLASLIB)
 	$(RANLIB) $@
 
 single: $(SMATGEN) $(SCATGEN)
