=== modified file 'gdb/blockframe.c'
--- gdb/blockframe.c	2010-11-12 17:13:12 +0000
+++ gdb/blockframe.c	2010-11-13 14:00:50 +0000
@@ -110,6 +110,40 @@
 	return fstart;
     }
 
+  /* Hacks to recognize the cintex trampolines. */
+  /* This only works on x86 for now. */
+  {
+    unsigned int code;
+    target_read_memory (pc+1, (char*)&code, sizeof(code));
+    if (code == 0xc920c483) {
+      /* add $0x20,%esp; leave; ret */
+      CORE_ADDR pc2 = pc - 0x20;
+      target_read_memory (pc2, (char*)&code, sizeof(code));
+      if (code == 0x83e58955) {
+        /* push $ebp; mov %esp, %ebp */
+        return pc2;
+      }
+      pc2 = pc - 0x1d;
+      target_read_memory (pc2, (char*)&code, sizeof(code));
+      if (code == 0x83e58955) {
+        /* push $ebp; mov %esp, %ebp */
+        return pc2;
+      }
+    }
+
+    /* Also for v21 trampolines. */
+    target_read_memory (pc-1, (char*)&code, sizeof(code));
+    if (code == 0xc3c9d0ff) {
+      /* call   *%eax; leave; ret */
+      CORE_ADDR pc2 = pc - 0x2f;
+      target_read_memory (pc2, (char*)&code, sizeof(code));
+      if (code == 0x83e58955) {
+        /* push $ebp; mov %esp, %ebp */
+        return pc2;
+      }
+    }
+  }
+
   return 0;
 }
 

